lib/components/DownloadsScreen/downloaded_items_list.dart

```diff
@@ -1,104 +1,168 @@
+class _DownloadedItemTypeListState extends ConsumerState<DownloadedItemsList> {
+  final _downloadsService = GetIt.instance<DownloadsService>();
+
+  final _supportedTypes = [
+    BaseItemDtoType.artist,
+    BaseItemDtoType.album,
+    BaseItemDtoType.playlist,
+    BaseItemDtoType.genre,
+    BaseItemDtoType.track,
+  ];
 
   @override
   Widget build(BuildContext context) {
-    var items = _downloadsService.getUserDownloaded();
-    return ListTileTheme(
-      // Manually handle padding in leading/trailing icons
-      horizontalTitleGap: 0,
-      child: SliverList(
-        delegate: SliverChildBuilderDelegate(
-          (context, index) {
-            DownloadStub album = items.elementAt(index);
-            return ExpansionTile(
-              key: PageStorageKey(album.id),
-              leading: Padding(
-                padding: const EdgeInsets.only(right: 16),
-                child: AlbumImage(item: album.baseItem),
-              ),
-              title: Text(album.baseItem?.name ?? album.name),
-              trailing: Row(
-                mainAxisSize: MainAxisSize.min,
-                children: [
-                  if ((!(album.baseItemType == BaseItemDtoType.album ||
-                          album.baseItemType == BaseItemDtoType.track)) &&
-                      !FinampSettingsHelper.finampSettings.isOffline)
-                    IconButton(
-                      icon: const Icon(Icons.sync),
-                      onPressed: () {
-                        _downloadsService.resync(album, null);
-                      },
-                    ),
-                  IconButton(
-                      icon: const Icon(Icons.delete),
-                      onPressed: () => askBeforeDeleteDownloadFromDevice(
-                              context, album, refresh: () {
-                            setState(() {});
-                          })),
-                ],
-              ),
-              subtitle: ItemFileSize(
-                stub: album,
-              ),
-              tilePadding: const EdgeInsets.symmetric(horizontal: 4.0),
-              children: [
-                DownloadedChildrenList(
-                  parent: album,
-                )
-              ],
-            );
-          },
-          childCount: items.length,
+    if (!_supportedTypes.contains(widget.type)) {
+      return const SliverToBoxAdapter(
+        child: Padding(
+          padding: EdgeInsets.symmetric(horizontal: 16, vertical: 8),
+          child: Text('Unsupported type'),
         ),
-      ),
-    );
+      );
+    }
+
+    return ref
+        .watch(_downloadsService.downloadedItemsProvider(widget.type))
+        .when(
+          data: (items) => items.isNotEmpty
+              ? SliverList(
+                  delegate: SliverChildBuilderDelegate(
+                    (context, index) {
+                      DownloadStub stub = items.elementAt(index);
+                      return ExpansionTile(
```

Comment:
Can you try to indent the child items when the tile is expanded? Currently the nested items look identical, except for the missing delete icon

Reply:
I can do that, but I personally prefer having the same indentation. Maybe we can otherwise separate them, e.g. by background color?

Reply:
Whatever you prefer! But BG color might be harder to do: changing the list items could bd messy, chaning the list BG requires passing (indentation) to look okay. A vertical left line/border with just a bit of indentation could work too?

Reply:
I like the line border idea. Let me try that then.

Reply:
Not sure I vibe with that:

<summary>Screenshot</summary>

![image](https://github.com/user-attachments/assets/38fbf8d5-e15b-4b85-aa4b-f8266a04cba5)
</details>

Reply:
A different color might work, some slightly lightened background color. But finding the right one there is hard.

Reply:
And you really don't like slightly indenting the items? :P

Reply:
I tried it and honestly wasn't a fan. But if there's no alternative we can also do that I guess.

Reply:
The background doesn't look too bad actually:
![image](https://github.com/user-attachments/assets/212d8e2d-2c91-4734-83df-c29859b71fba)
Using `surfaceVariant` since the other surface colors seem to be undefined in our theme, but fixing this is probably out of scope for this. For now, this color would work.

And here's the indentation version:
![image](https://github.com/user-attachments/assets/06b14048-f979-4be3-bce8-ac5ad68bf4b4)

Reply:
Fine, use the background for now, and when we get to redesigning the downloads screen we'll think about this again ^^  
Temporary victory for you :P